internal/codegen/parse.go

```diff
@@ -0,0 +1,140 @@
+	NameCamel      string
+	NamePascal     string
+	Optional       bool
+	Required       bool
+	Sensitive      bool
+	PointerString  string // either empty string or "Pointer" if property is optional
+	TypeString     string // "String" | "Float" ...
+	IsEnum         bool
+	EnumNameString string
+}
+
+type Verification struct {
+	Ref        string
+	ConfigRef  string
+	NameSnake  string
+	NameCamel  string
+	NamePascal string
+	NameConfig string // similar to NamePascal but for ThreeDEye it's 3DEye and other casing diff
+	Properties []VerificationProperty
+}
+
+func parseVerification(doc *openapi3.T, refName string) (*Verification, error) {
+	// Get verification data
+	verificationSchemaAny, err := doc.Components.Schemas.JSONLookup(refName)
+	if err != nil {
+		return nil, err
+	}
+	verificationSchema, ok := verificationSchemaAny.(*openapi3.Schema)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+	verificationName, ok := verificationSchema.Properties["type"].Value.Enum[0].(string)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+	verificationName = strings.ReplaceAll(verificationName, "-", "_")
+	docsType, ok := verificationSchema.Extensions["x-docs-type"].(string)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+
+	// Get verification config data
+	verificationConfigSchemaRef := verificationSchema.Properties["configs"].Ref
+	verificationConfigSchemaRefName := getSchemaNameFromRef(verificationConfigSchemaRef)
+
+	verificationConfigSchemaAny, err := doc.Components.Schemas.JSONLookup(verificationConfigSchemaRefName)
+	if err != nil {
+		return nil, err
+	}
+	verificationConfigSchema, ok := verificationConfigSchemaAny.(*openapi3.Schema)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+
+	// Constructing Verification struct
+
+	properties := []VerificationProperty{}
+
+	for _, key := range getProperties(verificationName, verificationConfigSchema) {
+		property := verificationConfigSchema.Properties[key]
+		required := slices.Contains(verificationConfigSchema.Required, key)
+		pointerString := ""
+		if !required {
+			pointerString = "Pointer"
+		}
+		typeString := ""
+		if property.Value.Type.Is("string") {
+			typeString = "String"
+		} else if property.Value.Type.Is("number") && property.Value.Format == "float" {
+			typeString = "Float64"
+		}
+		isEnum := false
+		enumNameString := ""
+		if len(property.Value.Enum) > 0 {
+			isEnum = true
+			if property.Ref == "" {
+				enumNameString = "New" + toConfigCase(verificationConfigSchemaRefName) + toConfigCase(key) + "From" + typeString
+			} else {
+				enumNameString = "New" + toConfigCase(getSchemaNameFromRef(property.Ref)) + "From" + typeString
+			}
+		}
+
+		properties = append(properties, VerificationProperty{
+			NameSnake:      key,
+			NameCamel:      strcase.ToLowerCamel(key),
+			NamePascal:     strcase.ToCamel(key),
+			Optional:       !required,
+			Required:       required,
+			Sensitive:      true,
```

Comment:
Just to reiterate, I'm defaulting to mark every property as "sensitive" because there's no way to tell which is sensitive and which is not from the schema POV. Here's the schema for one config, for example:

```json
"VerificationAPIKeyConfigs": {
  "type": "object",
  "properties": {
    "header_key": { "type": "string" },
    "api_key": { "type": "string" }
  },
  "required": ["header_key", "api_key"],
  "additionalProperties": false,
  "description": "The verification configs for API Key. Only included if the ?include=verification.configs query param is present",
  "x-docs-type": "APIKeyConfigs"
}
```

In this case, only `api_key` is sensitive but there's no way to tell.

One way to support this is add an extension to the properties like so

```json
"properties": {
  "header_key": { "type": "string", "x-sensitive": false },
  "api_key": { "type": "string", "x-sensitive": true }
}
```

This is doable from the server codebase. Let me know if you'd like to spend the effort here and I can help support that however I can.

Reply:
Reasonable to consider it all sensitive, that's what the server does as well we AES encrypt all the config